### PR TITLE
Added local file caching for better performance on many runs

### PR DESCRIPTION
**Pull request by oblodgett**

### CODE REVIEW

File: agr_indexer/src/main/java/org/alliancegenome/indexer/Main2.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.HashMap;
+
+import org.alliancegenome.core.config.ConfigHelper;
+import org.alliancegenome.curation_api.model.entities.AGMDiseaseAnnotation;
+import org.alliancegenome.curation_api.response.SearchResponse;
+import org.alliancegenome.indexer.indexers.curation.interfaces.AGMDiseaseAnnotationInterface;
+
+import si.mazi.rescu.RestProxyFactory;
+
+public class Main2 {
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/Main2.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.design.HideUtilityClassConstructorCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Utility classes should not have a public or default constructor.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/BaseDiseaseAnnotationService.java [hunk 2]
```diff
@@ -107,14 +107,14 @@ protected boolean hasNoObsoletedOrInternalEntities(DiseaseAnnotation da) {
 			entitiesToBeValidated.addAll(da.getDiseaseGeneticModifiers());
 		}
 		AtomicBoolean hasNoObsoletedOrInternalEntities = new AtomicBoolean(true);
-		entitiesToBeValidated.forEach(auditedObject -> {
+		for(AuditedObject auditedObject: entitiesToBeValidated) {
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/BaseDiseaseAnnotationService.java:110]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' is not followed by whitespace.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/GeneDiseaseAnnotationService.java [hunk 3]
```diff
@@ -87,6 +91,18 @@ public List<GeneDiseaseAnnotation> getFiltered() {
 	public Map<Gene, List<DiseaseAnnotation>> getOrthologousGeneDiseaseAnnotations(Map<String, Pair<Gene, ArrayList<DiseaseAnnotation>>> geneMap) {
 		ProcessDisplayHelper display = new ProcessDisplayHelper(10000);
 
+		String orthoCacheFileName = "gene_disease_via_orthology_annotation.json.gz";
+		
+		HashMap<Gene, List<DiseaseAnnotation>> newDAMap = readFromCache(orthoCacheFileName, HashMap.class);
+		
+		if(newDAMap == null) {
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/GeneDiseaseAnnotationService.java:98]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/GeneDiseaseAnnotationService.java [hunk 4]
```diff
@@ -117,7 +133,9 @@ public Map<Gene, List<DiseaseAnnotation>> getOrthologousGeneDiseaseAnnotations(M
 					continue;
 				}
 				// create orthologous DAs for each focus DA
-				focusDiseaseAnnotations.forEach(focusDiseaseAnnotation -> {
+				
+				for(DiseaseAnnotation focusDiseaseAnnotation: focusDiseaseAnnotations) {
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/GeneDiseaseAnnotationService.java:137]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' is not followed by whitespace.

File: agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java [hunk 5]
```diff
@@ -157,263 +177,295 @@ public void run() {
...
 		@Override
 		public void run() {
+
 			log.info("Building gene -> cross references map");
-			String query = "MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference) ";
-			query += " RETURN gene.primaryKey as id, cr.name as value";
+			try {
 
-			Map<String, Set<String>> names = getMapSetForQuery(query);
+				String query = "MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference) ";
```
Comment (cmpich) [inline agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java]: Can you use text blocks that makes it more readable and usable. You can copy / paste the whole query in one step.
"""
MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference)
RETURN gene.primaryKey as id, c
"""

File: agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java [hunk 6]
```diff
@@ -132,23 +136,39 @@ public void setCacheResult(Map<String, Set<String>> result) {
...
 			Map<String, Gene> geneMap = new HashMap<>();
-			for (Gene gene : genes) {
-				geneMap.put(gene.getPrimaryKey(), gene);
+
+			try {
+				if (!cacheFile.exists()) {
+
+					String query = " MATCH p1=(species:Species)-[:FROM_SPECIES]-(g:Gene) ";
+					query += " OPTIONAL MATCH pSoTerm=(g:Gene)-[:ANNOTATED_TO]-(soTerm:SOTerm)";
+					query += " OPTIONAL MATCH p5=(g:Gene)--(:GenomicLocation) ";
```
Comment (cmpich) [inline agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java]: Please use text block (triple quotes) for readability.

Comment (oblodgett) [reply agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java]: I was not making a change to the queries here. And I agree we do need to go through the code and make it more readable using code chunks.

**Pull request merged**